### PR TITLE
Added Stack [C#]

### LINKED ISSUE #501: Stack [C#]
#### This issue is about [Stack.](http://www.cdn.geeksforgeeks.org/stack-data-structure-introduction-program/)

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by franetheman**
Fixes #501 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

File: stack/Stack.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    */
+    public bool IsEmpty()
+    {
+        return (!stack.Any());
+    }
+
+    /**
+    * Swaps places between the last and second last element of the stack
+    */
+    public void Swap()
```
Comment (singhpratyush) [inline stack/Stack.cs]: Is this method needed?

File: stack/Stack.cs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        if (IsEmpty())
+            return default(T);
+
+        return stack.Last();
+    }
+
+    /**
+    * pops and pushes the same element twice
+    */
+    public void Duplicate()
```
Comment (singhpratyush) [inline stack/Stack.cs]: Same here. Do we need such a method? Is it implemented in any of the standard libraries?

**Comment by franetheman** [comment]
I'll remove those two since they're not essential for the data structure to function

File: stack/Stack.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        }
+    }
+}
+
+class Stack<T>
+{
+    private LinkedList<T> stack;
+
+    /**
+    * Constructs an empty stack
```
Comment (mohitkyadav) [inline stack/Stack.cs]: I think one line is enough for this comment.

File: stack/Stack.cs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    */
+    public void Push(T data)
+    {
+        stack.AddLast(data);
+    }
+
+    /**
+    * Removes the element at the top of the stack
+    */
+    public void Pop()
```
Comment (singhpratyush) [inline stack/Stack.cs]: Also, the pop method should return the popped data.

File: stack/Stack.cs [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        Console.WriteLine(test.ToString());
+
+        Console.WriteLine(test.Peek() + "\n");
+
+        while (!(test.IsEmpty()))
+        {
+            test.Pop();
+            Console.WriteLine(test.ToString());
+        }
+
```
Comment (singhpratyush) [inline stack/Stack.cs]: Build fails due to trailing whitespace.

**Pull request merged**